test/index.js

```diff
@@ -935,62 +832,94 @@ describe('Mongo', () => {
+
+            const options = {
+                uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
+                partition: 'unit-testing'
+            };
+            const key = {
+                id: 'testerr',
+                segment: 'testerr'
+            };
+            const mongo = new Mongo(options);
+            mongo.isConnectionStarted = true;
+            mongo.isConnected = false;
+
+            mongo.collections.testerr = {
+                findOne: function (item) {
+
+                    return Promise.resolve({ value: false });
+                }
+            };
+
+            try {
+                const result = await mongo.get(key);
+                expect(result).to.not.exist();
+            }
+            catch (err) {
+                expect(err).to.exist();
+                expect(err.message).to.equal('Connection not ready');
+            }
+        });
     });
 
     describe('set()', () => {
 
-        it('passes an error to the callback when the connection is closed', (done) => {
+        it('returns an error when the connection is closed', async () => {
 
             const options = {
                 uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
 
-            mongo.set({ id: 'test1', segment: 'test1' }, 'test1', 3600, (err) => {
-
+            try {
+                await mongo.set({ id: 'test1', segment: 'test1' }, 'test1', 3600);
+            }
+            catch (err) {
                 expect(err).to.exist();
                 expect(err).to.be.instanceOf(Error);
                 expect(err.message).to.equal('Connection not started');
-                done();
-            });
+            }
         });
 
-        it('doesn\'t return an error when the set succeeds', (done) => {
+        it('doesn\'t return an error when the set succeeds', async () => {
 
             const options = {
                 uri: 'mongodb://127.0.0.1:27017/?maxPoolSize=5',
                 partition: 'unit-testing'
             };
             const mongo = new Mongo(options);
 
-            mongo.start(() => {
-
-                mongo.set({ id: 'test1', segment: 'test1' }, 'test1', 3600, (err, result) => {
+            await mongo.start();
 
-                    expect(err).to.not.exist();
-                    expect(result).to.not.exist();
-                    done();
-                });
-            });
+            try {
+                const result = await mongo.set({ id: 'test1', segment: 'test1' }, 'test1', 3600);
+                expect(result).to.not.exist();
+            }
+            catch (err) {
+                expect(err).to.not.exist();
+            }
         });
 
-        it('passes an error to the callback when there is an error returned from setting an item', (done) => {
+        it('passes an error to the callback when there is an error returned from setting an item', async () => {
```

Comment:
I noticed that you renamed many of the tests so it correctly describes that it does - but there is still several old ones talking in callback terms.